server/test/support.ts

```diff
@@ -7,17 +11,95 @@ import onerror from "@server/onerror";
 import webService from "@server/services/web";
 import { sequelize } from "@server/storage/database";
 
+class TestServer {
+  private server: http.Server;
+  private listener?: Promise<void> | null;
+
+  constructor(app: Koa) {
+    this.server = http.createServer(app.callback() as any);
+  }
+
+  get address(): string {
+    const { port } = this.server.address() as AddressInfo;
+    return `http://localhost:${port}`;
+  }
+
+  listen() {
+    if (!this.listener) {
+      this.listener = new Promise((resolve, reject) => {
+        this.server
+          .listen(0, () => resolve())
+          .on("error", (err) => reject(err));
+      });
+    }
+
+    return this.listener;
+  }
+
+  fetch(path: string, opts: any) {
```

Comment:
It's actually quite nice having this helper internally now, will probably consider having another argument here for `user` and then change all the tests so that the jwtToken doesn't need to be injected into the headers everywhere?

(Obviously not in this PR)